File: Bymed.Application/CatalogueItems/CatalogueItemDto.cs
namespace Bymed.Application.CatalogueItems;

public sealed record CatalogueItemDto
{
    public required Guid Id { get; init; }
    public required string Name { get; init; }
    public required string Slug { get; init; }
    public required string Description { get; init; }
    public required Guid CategoryId { get; init; }
    public required string CategoryName { get; init; }
    public string? PrimaryImageUrl { get; init; }
    public IReadOnlyList<CatalogueItemImageDto>? Images { get; init; }
    public bool IsPublished { get; init; }
    public Guid? BrandId { get; init; }
    public string? BrandName { get; init; }
    public string? BrandLogoUrl { get; init; }
    public string? BrandWebsiteUrl { get; init; }
}

File: bymed-admin/src/app/shared/models/catalogue-item.dto.ts
export interface CatalogueItemImageDto {
  readonly id: string;
  readonly catalogueItemId: string;
  readonly url: string;
  readonly altText: string;
  readonly displayOrder: number;
}

export interface CatalogueItemDto {
  readonly id: string;
  readonly name: string;
  readonly slug: string;
  readonly description?: string;
  readonly categoryId: string;
  readonly categoryName?: string;
  readonly primaryImageUrl?: string;
  readonly images?: CatalogueItemImageDto[];
  readonly isPublished: boolean;
  readonly brandId?: string | null;
  readonly brandName?: string | null;
  readonly brandLogoUrl?: string | null;
  readonly brandWebsiteUrl?: string | null;
}

export interface CreateCatalogueItemRequestDto {
  readonly name: string;
  readonly description: string;
  readonly categoryId: string;
  readonly brandId?: string | null;
  readonly isPublished: boolean;
}

export interface UpdateCatalogueItemRequestDto {
  readonly name: string;
  readonly description: string;
  readonly categoryId: string;
  readonly brandId?: string | null;
  readonly isPublished: boolean;
}
